--- yosys.orig/Makefile
+++ yosys/Makefile
@@ -21,6 +21,7 @@
 ENABLE_LIBYOSYS := 0
 ENABLE_PROTOBUF := 0
 ENABLE_ZLIB := 1
+ENABLE_GHDL := 0
 
 # python wrappers
 ENABLE_PYOSYS := 0
@@ -81,7 +82,8 @@
 YOSYS_SRC := $(dir $(firstword $(MAKEFILE_LIST)))
 VPATH := $(YOSYS_SRC)
 
-CXXFLAGS := $(CXXFLAGS) -Wall -Wextra -ggdb -I. -I"$(YOSYS_SRC)" -MD -D_YOSYS_ -fPIC -I$(PREFIX)/include
+# CXXFLAGS := $(CXXFLAGS) -Wall -Wextra -ggdb -I. -I"$(YOSYS_SRC)" -MD -D_YOSYS_ -fPIC -I$(PREFIX)/include
+CXXFLAGS := $(CXXFLAGS) -Wall -Wextra -g -I. -I"$(YOSYS_SRC)" -MD -D_YOSYS_ -fPIC -I$(PREFIX)/include
 LDLIBS := $(LDLIBS) -lstdc++ -lm
 PLUGIN_LDFLAGS :=
 
@@ -183,7 +185,7 @@
 ifeq ($(CONFIG),clang)
 CXX = clang
 LD = clang++
-CXXFLAGS += -std=c++11 -Os
+CXXFLAGS += -std=c++11 # -Os
 ABCMKARGS += ARCHFLAGS="-DABC_USE_STDINT_H"
 
 ifneq ($(SANITIZER),)
@@ -523,6 +525,16 @@
 ifeq ($(ENABLE_PROTOBUF),1)
 LDLIBS += $(shell pkg-config --cflags --libs protobuf)
 endif
+
+ifeq ($(ENABLE_GHDL),1)
+GHDL_DIR ?= /usr/local/ghdl
+GHDL_INCLUDE_DIR ?= $(GHDL_DIR)/include
+GHDL_LIB_DIR ?= $(GHDL_DIR)/lib
+CXXFLAGS += -I$(GHDL_INCLUDE_DIR) -DYOSYS_ENABLE_GHDL
+LDLIBS += $(GHDL_LIB_DIR)/libghdlsynth.a $(GHDL_LIB_DIR)/libghdl.a $(shell cat $(GHDL_LIB_DIR)/ghdlsynth.link)
+endif
+
+LDLIBS += $(EXTRA_LIBS)
 
 ifeq ($(ENABLE_COVER),1)
 CXXFLAGS += -DYOSYS_ENABLE_COVER
